certification/internal/policy/container/has_modified_files.go

```diff
@@ -76,9 +97,20 @@ func (p *HasModifiedFilesCheck) getDataToValidate(ctx context.Context, imgRef ce
 	return &packageFilesRef{files, packageFiles}, nil
 }
 
+// validate compares the list of LayerFiles and PackageFiles to see what PackageFiles
+// have been modified within the additional layers.
 func (p *HasModifiedFilesCheck) validate(packageFilesRef *packageFilesRef) (bool, error) {
 	layerFiles := packageFilesRef.LayerFiles
 	packageFiles := packageFilesRef.PackageFiles
+
+	// In cases where the image was built FROM scratch,
+	// we drop the empty layer. This should be relatively rare,
+	// but it is the case in images such as ubi-micro.
+	if len(layerFiles[0]) == 0 {
```

Comment:
Would it be better to leave this method as is, and perhaps drop the layer in the gather method? Since we have the cranev1.Layer? But now I see that layer.Size() is the *compressed* size... So that's not exactly what we want... is it possible to have a single zero-length file in a layer and have it's size be zero?

Reply:
It doesn't seem like that would be possible... But, I think we can rely on the Compressed() size to be 32. And that the hash would be 4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1. So, we could check for either of those things. Very magic-numbery, but I think they both can be relied upon. I just feel like validate shouldn't need to massage the data any further. That's the job of getDataToValidate. Feel free to ignore me.

Reply:
I moved it to the getDataToValidate method, though probably not in the cleanest way. I just mirrored the exact behavior, just in this function instead of in the Validate function - only because I can't seem to test against UAT at the moment. Local tests pass, and the logic should accomplish the same thing. The `packageFilesRef` goes untouched between the two functions.

Reply:
Also, for what it's worth, I kept the check as designed (check for slice for a non-zero length, indicating that files were populated in that layer) for the moment, only because we don't ever want to encounter zero files in the first layer (as opposed to relying on the compressed size or the hash, which may or may not be variable). 

The way I see it, if our logic iterates over the first value in that slice, `packageFilesRef.LayerFiles`, then we should just explicitly check to make sure that's non-zero.

I think it may be possible to find a case where this... doesn't quite work how I think it should, but I'll need to spend cycles testing on it and come back to it. For the moment, though, with our UBI images, it should cover the bases.